### ISSUE TITLE
New feature that allows inserting named-destination links doesn't work

### ISSUE THREAD

**Issue by jackb007** [pymupdf/PyMuPDF#2922]
### Description of the bug

The xref of the link gets set to 0 even if it exists.

```
import fitz

def show_bug(doc):
	for index, page in enumerate(doc):
		print()
		print(index)
		links = page.get_links()

		if not links:
			continue

		link = links[0]
		print(link)
		page.delete_link(link)
		page.insert_link(link)
		links = page.get_links()
		link = links[0]
		print(link)

path = 'C:\\path\\to\\doc-with-working-named-destination-links.pdf'
doc = fitz.Document(path)
show_bug(doc)
doc.close()
```

### How to reproduce the bug

How to reproduce the bug: run the bug-revealing script provided above on a pdf document containing named-destination links.

Sample output of the bug-revealing script:

{'kind': 4, 'xref': 2395, 'from': Rect(266.02301025390625, 387.9209899902344, 273.0050048828125, 399.0389709472656), 'name': 'nameddest=prob01.chap1.label', 'id': ''}
{'kind': 4, 'xref': 0, 'from': Rect(266.02301025390625, 387.9209899902344, 273.0050048828125, 399.0389709472656), 'name': 'nameddest=prob01.chap1.label'}

Note how the xref has disappeared. The links in the original file are working perfectly fine. An exact copy with the links deleted and immediately re-inserted contains links that don't work (due to the missing xref?).

### PyMuPDF version

1.23.8

### Operating system

Windows

### Python version

3.9

**Comment by julian-smith-artifex-com** [comment]
This is fixed in release candidate 1.23.9rc1; see: #2960

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.9.
